### ISSUE TITLE
google_sql_database_instance fails creation when activation_policy is NEVER

### ISSUE THREAD

**Issue by wastrachan** [hashicorp/terraform-provider-google#8874]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
v0.14.9

### Affected Resource(s)
google_sql_database_instance

### Terraform Configuration Files

```tf
resource "google_sql_database_instance" "database" {
  name                = "test-db"
  database_version    = "MYSQL_5_7"
  region              = "us-east4"
  deletion_protection = false

  settings {
    tier              = "db-custom-12-73728"
    disk_autoresize   = true
    disk_type         = "PD_SSD"
    availability_type = "ZONAL"
    activation_policy = "NEVER"
    }
}
```

### Debug Output

https://gist.github.com/wastrachan/b6872e3d199cdff136c6d62fdcdc25cc

This debug output is abbreviated and partially redacted for privacy. If required, I will provide a complete output signed with Hashicorp's GPG key.

### Expected Behavior

I would expect one of two things to occur:

1. Database to be created with an `activation_policy` of `NEVER` (stopped)
2. An error message indicating that `NEVER` is not an acceptable `activation_policy` for new resources, and a documentation caveat confirming this.

### Actual Behavior

`terraform apply` failed with a generic error:

```
module.backups_recovery_db.google_sql_database_instance.database: Creating...

Error: Error, failed to create instance backups-recovery: googleapi: Error 400: This operation is not valid for this instance., invalidOperation

  on ../../modules/sql/main.tf line 3, in resource "google_sql_database_instance" "database":
   3: resource "google_sql_database_instance" "database" {
```

### Steps to Reproduce

1. Create a `google_sql_database_instance` resource with `settings.activation_policy = "NEVER"`
2. `terraform apply` 

So far as I have been testing, the component settings of the database instance do not seem to have an impact on the outcome.

**Comment by venkykuberan** [comment]
@wastrachan That's the API behavior, provider can't control it. Were you able to create the sql database instance outside of terraform (REST Call) with the same settings ?. If you observe different behavior there, please pass it on to us. We will take it from there.

**Comment by wastrachan** [comment]
> @wastrachan That's the API behavior, provider can't control it. Were you able to create the sql database instance outside of terraform (REST Call) with the same settings ?. If you observe different behavior there, please pass it on to us. We will take it from there.

No, not able to get different behavior with a direct API call. 

I totally understand it's API behavior. I think the issue (which is probably more of an enhancement request now, considering) is that the provider has an opportunity to validate and provide a more consistent error message to the end-user. I spent an hour chasing `Error 400: This operation is not valid for this instance.` before I realized what was going on. 

Barring validation, a note in documentation also might help anybody trounced by Google's unhelpful error message in the future. 

I've got no expectation that the provider would be able to create a database in a stopped state, but I did want to try to save anybody who might try some heartburn.

**Comment by venkykuberan** [comment]
Got you, We can add validation logic for that attribute. I will mark this as an enhancement request, Thank you for filing it.

**Comment by slevenick** [comment]
https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances#Settings
The activation policy specifies when the instance is activated; it is applicable only when the instance state is RUNNABLE. Valid values:
ALWAYS: The instance is on, and remains so even in the absence of connection requests.
NEVER: The instance is off; it is not activated, even if a connection request arrives.

We could implement this as a post-create step

**Comment by melinath** [comment]
It looks like this might be resolved? b/259900758
